File: 3147-taking-maximum-energy-from-the-mystic-dungeon/3147-taking-maximum-energy-from-the-mystic-dungeon.cpp
class Solution {
public:
    int maximumEnergy(vector<int>& energy, int k) {
        int n=energy.size(),maxi = INT_MIN;
        for(int i=n-1;i>=0;i--){
            if(i+k<n){
                energy[i] += energy[i+k];
            }
            maxi = max(maxi,energy[i]);
        }
        return maxi;
    }
};

File: 3147-taking-maximum-energy-from-the-mystic-dungeon/3147-taking-maximum-energy-from-the-mystic-dungeon.java
class Solution
{
    public int maximumEnergy(int[] energy, int k)
    {
        int l = energy.length;
        
        for(int i=l-k-1; i>=0; i--)
            energy[i] += energy[i+k];

        int r = Integer.MIN_VALUE;

        for(int i=0; i<l; i++)
            r = Math.max(r,energy[i]);

        return r;
    }
}